src/utils/sqs_consumer.ts

```diff
@@ -0,0 +1,91 @@
+import { SQS } from 'aws-sdk';
+import delay from 'delay';
+
+import { ONE_SECOND_MS } from '../constants';
+import { logger } from '../logger';
+
+import { SqsClient } from './sqs_client';
+
+export type MessageHandler = (message: SQS.Types.Message) => Promise<any>;
+
+export class SqsConsumer {
+    private readonly _sqsClient: SqsClient;
+    private readonly _beforeHandle?: () => Promise<boolean>;
+    private readonly _handleMessage: MessageHandler;
+    private readonly _afterHandle?: MessageHandler;
+    private _isConsuming: boolean;
+
+    constructor(params: {
+        sqsClient: SqsClient;
+        beforeHandle?: () => Promise<boolean>;
+        handleMessage: MessageHandler;
+        afterHandle?: MessageHandler;
+    }) {
+        this._sqsClient = params.sqsClient;
+        this._beforeHandle = params.beforeHandle;
+        this._handleMessage = params.handleMessage;
+        this._afterHandle = params.afterHandle;
+        this._isConsuming = false;
+    }
+
+    public stop(): void {
+        this._isConsuming = false;
+    }
+
+    public async consumeAsync(): Promise<void> {
+        if (this._isConsuming) {
+            return;
+        }
+
+        this._isConsuming = true;
+        while (this._isConsuming) {
+            await this.consumeOnceAsync();
+        }
+    }
+
+    public async consumeOnceAsync(): Promise<void> {
```

Comment:
should this be private?

Reply:
I made it public so I could test it more easily. Also figured it might be useful to have a consumer where someone manually handles the consumption as well (maybe they want to consume a predefined number of messages)

Reply:
I see.. so it's public for testing purposes right?

Reply:
correct! Although I could see it being useful by itself as well